SOPT30th-1st-Assignment/SOPT30th-1st-Assignment/ThirdViewController.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  ThirdViewController.swift
+//  SOPT30th-1st-Assignment
+//
+//  Created by Zaid Kang on 2022/04/08.
+//
+
+import UIKit
+
+class ThirdViewController: UIViewController {
+    var name: String?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+    }
+    
+
+    @IBAction func moveFourthController(_ sender: Any) {
+        guard let nextVC = self.storyboard?.instantiateViewController(withIdentifier: "FourthViewController") as? FourthViewController else { return }
+        
+        guard let name = name else { return }
+        nextVC.name = "\(name)님, 인스타그램에 오신것을 환영합니다."
+        nextVC.navigation = self.navigationController
+        
+        self.present(nextVC, animated: true)
```

Comment:
> 완료 화면에서 로그인 화면으로 dismiss 하기 위해 popToRootViewController 메소드를 사용했는데 동작하지 않았습니다. 이전 view에서의 전달 방식이 present 방식이라 Navigation Controller가 없어서 그런거라고 해서 ThirdViewController에서 프로퍼티에 주입하여 구현하였습니다.
```suggestion
        self.present(nextVC, animated: true) {
            self.navigationController?.popToRootViewController(animated: true)
        }
```
`self.present()`의 실행이 완료된 후에 실행되는 completionHandler를 이용하면 present가 된후에 -> navigationController의 stack을 pop시키면 -> 정상적으로 작동합니다!!

이렇게 해주면 navigationController를 따로 주입해주지 않아도 되겠네요!!